cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -51,8 +50,6 @@ func newAsgCache(asgCacheTTL int64) (*asgCache, error) {
 	}
 
 	go wait.Until(func() {
-		cache.mutex.Lock()
-		defer cache.mutex.Unlock()
 		if err := cache.regenerate(); err != nil {
```

Comment:
I think we should also get rid of this. This TTL is useless with the changes here/we already refresh the instances cache way faster than this.